### DISCUSSION TITLE
KAMP prints purge line under supports in Orca

### DISCUSSION THREAD

**Discussion by shirgall** [Guilouz/Creality-Helper-Script-Wiki#549] [Issues]
K1 Max w.1.3.3.26 firmware, factory install, w/script+enough stuff to run KAMP:

Some of my complex objects need tall support, which Orca w/auto Trees with default settings go all the way to the bed, but when those supports go "south" of the object, they can intersect with the KAMP purge line, leading to poor integrity for that support or set of supports.

<img width="755" alt="image" src="https://github.com/Guilouz/Creality-Helper-Script-Wiki/assets/3824421/dcd82739-3e2e-4ce9-87fc-26e8ca65ea10">

I can rotate the object and not have an issue, but I figured I should document this in case others have seen it.

**Comment by shirgall** [comment]
"KAMP uses information from [[exclude_object]](https://www.klipper3d.org/Exclude_Object.html#exclude-objects) to define mesh bounds. Complicated slicer parameter passing is finally a thing of the past. Welcome to the future!"

Perhaps this is an issue with exclude_object parameters?

**Reply by shirgall** [reply]
I discovered I did not have "Label Objects" enabled, but enabling that did not correct the problem.
<img width="402" alt="image" src="https://github.com/Guilouz/Creality-Helper-Script-Wiki/assets/3824421/b6a041be-7b15-4690-8ce7-9ab954bc53b6">

**Reply by Guilouz** [reply]
Read the wiki.

**Reply by shirgall** [reply]
Yeah, I have, but I've not had to adjust the margins before. Either because I factory reset and installed everything fresh with the new firmware, but I've never needed to set the variable mesh margin before. Live and learn. Thanks for all of the scripts!

**Reply by robomopa** [reply]
I also got this problem. Is it possible to set the classic purge line default, so users that dont want the adaptive purge wont need to select from fluidd after each power-on/off cycle

**Comment by JooJooBee666** [comment]
I have the solution!

In Fluid, open the configuration menu and browse to the **/Helper-Script/KAMP** folder and edit the **KAMP_Settings.cfg** file.  Look for the following line:
`variable_purge_margin: 10                   # Distance the purge will be in front of the print area. Default is 10.`
and change it to something like:
`variable_purge_margin: 50                   # Distance the purge will be in front of the print area. Default is 10.`
B00M! No more tree base f**kery!

**Reply by dmchmk** [reply]
ka-boom, print bed border f**kery((
